haystack/backends/solr_backend.py

```diff
@@ -296,6 +297,15 @@ def build_search_kwargs(
                     gap_by_string,
                 )
 
+        if interval_facets is not None:
+            kwargs["facet"] = "on"
+            kwargs["facet.interval"] = interval_facets.keys()
```

Comment:
```suggestion
            kwargs["facet.interval"] = list(interval_facets)
```
Might be slightly safer...
```
>>> d = {key: value for value, key in enumerate("ABC")}
>>> d
{'A': 0, 'B': 1, 'C': 2}
>>> d.keys()[1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_keys' object is not subscriptable
>>> list(d)[1]
'B'
```

Reply:
Makes sense, although I'm not sure that it would ever be called that way (since you'd only use the `dict`), at the same time I don't know if there's a reason for it to a a `dict_keys` view either. I think technically `.keys()` is quicker since it's not making a new list, but in this case I can't see that ever being an issue. Happy to adjust, it seems to work either way on my running build.